File: examples/python/basic_connect.py
#!/usr/bin/env python3
"""
Example: Connect to Vouch Daemon and Check Status

This is the simplest example - just connect and see if the daemon is running.
"""

from vouch_sdk import VouchClient, VouchConnectionError


def main():
    # Create a client (uses default localhost:21000)
    client = VouchClient()

    try:
        # Connect to the daemon
        status = client.connect()

        print("✅ Vouch Daemon is online!")
        print(f"   Version: {status.get('version', 'unknown')}")
        print(f"   Has Keys: {status.get('has_keys', False)}")
        print(f"   Uptime: {status.get('uptime', 0)} seconds")

    except VouchConnectionError as e:
        print("❌ Vouch Daemon is offline!")
        print(f"   Error: {e}")
        print("\n💡 Start the daemon with: vouch-bridge")
        return False

    return True


if __name__ == "__main__":
    main()

File: examples/typescript/basic-connect.ts
/**
 * Example: Connect to Vouch Daemon
 *
 * The simplest example - check if the daemon is running.
 */

import { VouchClient, VouchConnectionError } from '@vouch-protocol/sdk';

async function main() {
    // Create a client (uses default localhost:21000)
    const client = new VouchClient();

    try {
        // Connect to the daemon
        const status = await client.connect();

        console.log('✅ Vouch Daemon is online!');
        console.log(`   Version: ${status.version || 'unknown'}`);
        console.log(`   Has Keys: ${status.has_keys}`);

    } catch (error) {
        if (error instanceof VouchConnectionError) {
            console.log('❌ Vouch Daemon is offline!');
            console.log('   Start the daemon with: vouch-bridge');
        } else {
            throw error;
        }
    }
}

main();
